facet-peek/src/enum_.rs

```diff
@@ -138,19 +138,20 @@ impl<'mem> PeekEnum<'mem> {
     /// Returns a Peek handle to a field of a tuple or struct variant
     pub fn field(self, field_name: &str) -> Option<crate::Peek<'mem>> {
         let variant = self.active_variant();
+        let variant_offset = variant.offset;
 
         match &variant.kind {
             VariantKind::Unit => None, // Unit variants have no fields
             VariantKind::Tuple { fields } => {
                 // For tuple variants, find by name
                 let field = fields.iter().find(|f| f.name == field_name)?;
-                let field_data = unsafe { self.value.data().field(field.offset) };
+                let field_data = unsafe { self.value.data().field(variant_offset + field.offset) };
```

Comment:
I don't love this -- super easy to miss the variant offset and create unsoundness. But it seemed consistent with what there is today.

I _think_ it would be nice if selecting a variant shifted the `self.data` offset appropriately perhaps?

Reply:
Pushed https://github.com/facet-rs/facet/pull/157/commits/c4be0ef988b24af13d6d534bdff18081aa891a58 which I think is an improvement and formalises a little what's going on here -- all our enum variant field offsets are computed relative to the start of the "variant data" section. For `#[repr(u*)]` that variant data offset is 0 and each variant basically adds padding for the discriminant. For `#[repr(C)]` the variant data offset should be fixed and non-zero, but fields in a variant can start from zero.